apps/src/lab2/views/Lab2ShareDialogWrapper.tsx

```diff
@@ -51,22 +61,29 @@ const Lab2ShareDialogWrapper: React.FunctionComponent<
   const isPublished = false;
   const canShareSocial = isSignedIn && is13Plus;
 
+  const onCloseSubmitProjectDialog = useCallback(() => {
+    dispatch(setShowSubmitProjectDialog(false));
+  }, [dispatch]);
+
   if (!channelId || !projectType) {
     return null;
   }
 
   if (LABS_USING_NEW_SHARE_DIALOG.includes(projectType)) {
-    if (!isOpen) {
-      return null;
-    }
-
     return (
-      <ShareDialog
-        dialogId={dialogId}
-        shareUrl={shareUrl}
-        finishUrl={finishUrl}
-        projectType={projectType}
-      />
+      <>
+        {showSubmitProjectDialog && (
+          <SubmitProjectDialog onClose={onCloseSubmitProjectDialog} />
+        )}
+        {isShareDialogOpen && (
```

Comment:
showSubmitProjectDialog` and `isShareDialogOpen` are rather asymmetric.  Could we have consistent naming of these conditions, and moreover does this hint that we could unify into a single state which captures _which_ dialog is currently showing?  

Ideally we'd have a centralized state tracking which dialog is showing, rather than a collection of independent dialogs taking care of showing themselves, which could lead to state conflicts if both of them accidentally start showing simultaneously.

Reply:
+1, this might be more easily managed as internal state here or within `ShareDialog`

Reply:
Thanks! Will take another pass with these suggestions in mind.

Reply:
Hmmm... after a refactor, when the share dialog is rendered, it is first displayed without the 'Submit to be featured' button, and then with it. 

Any ideas on how to avoid this? Screencast below.

https://github.com/user-attachments/assets/24b44d70-ee1a-4aad-9899-ad9b7b1406a7

Reply:
I moved the fetching of submission status to `Lab2ShareDialogWrapper` and that helped. Now the rendering without the 'Submit to be featured' button happens only once on initial modal render.

https://github.com/user-attachments/assets/fd17ab52-f32c-4fe3-bda6-ec6c50ab72c8

Reply:
Nice! This feels cleaner. However, this does still have a bit of the original issue that Brendan mentioned, in that the two pieces of state that control which dialog to show are technically independent. If hypothetically some other system were to set `isShareDialogOpen` to true while the submit dialog was open (since it's globally available via redux), then we'd end up with both dialogs open.

One suggestion would be to consolidate the state of which specific panel of the Share Dialog is showing; for example something like:
```
const [dialogPane, setDialogPane] = useState<'share' | 'submit'>('share'); // state to track which specific dialog is showing (can be named something different)

const onSubmitClick = () => setDialogPane('submit'); // clicking submit takes us to the submit dialog
const onGoBack = () => setDialogPane('share'); // clicking back in submit takes us back to share 

....

return {
  ...
  dialogPane === 'share' ? <ShareDialog .../> : <SubmitDialog ... />;
  
}
```

That way the internal `dialogPane` (or better name) state completely controls the internals of the dialog state once its open, and the redux state `isShareDialogOpen` completely controls only entering and exiting the overall dialog system. `isShareDialogOpen` does become a slightly incorrect name then, but I think separating these two pieces of state helps keeps things more consistent and stable.

Reply:
Thanks Sanchit! I tried out your suggestion https://github.com/code-dot-org/code-dot-org/pull/62069/commits/70172fb76632b0f2ff639ebbc98bbf0d8022e3ef, and got some unexpected behavior. When I click on 'Go back', the `goBack` function is called, but so is the `onClose` function in `SubmitProjectDialog` so that click on 'Go back' not only closes the 'submit' dialog but the 'share' dialog is also not displayed.
Maybe this has to do with `AccessibleDialog` which I'm using in `SubmitProjectDialog`?

Reply:
I believe this does have to do with`AccessibleDialog`. `FocusTrap` has an `onDeactivate` option which is set to `onClose`. https://github.com/code-dot-org/code-dot-org/blob/4e1d5e8cd0f537c20df8313da93ee423a0b5bd12/apps/src/sharedComponents/AccessibleDialog.jsx#L39
If I remove this option, the submit project dialog works as expected. I think I’ll need to prep a PR that will allow this to be configurable.

Reply:
`AccessibleDialog` PR: https://github.com/code-dot-org/code-dot-org/pull/62228